src/context/WalletProvider/WalletProvider.tsx

```diff
@@ -180,14 +181,24 @@ export const WalletProvider = ({ children }: { children: React.ReactNode }): JSX
     }
   }, [])
 
+  const create = useCallback(async (type: KeyManager) => {
+    dispatch({ type: WalletActions.SET_CONNECTOR_TYPE, payload: type })
+    if (SUPPORTED_WALLETS[type]?.routes[5]?.path) {
```

Comment:
i'd much prefer if we index into this by `routes.find(path => path.endsWith('create'))` - there's nothing in that config file that suggests ordering is important and this can and will break in funky ways

Reply:
It's a pre-existing pattern though, so if we make this change we should also update the `connect` route to match.

Reply:
ah you're right, but yeah, @evandrosaturnino if you wouldn't mind making that small change in `connect` too - this code was rushed out at the 11th hour originally and could do with some love.

Reply:
Applied the requested changes both in `create` and `connect`, set the path into String() method, once `.endsWith()` can't be applied in routeprops type.
Unit tests are passing as expected.
Changes were submitted on commit 911e67ea66d46b00ca2497c634bc36d7a30de090

Reply:
Nice work.

Here is a fun little optimisation of that logic for brevity and concision, if you're interested @evandrosaturnino 🙂

We can use `lodash`'s `findIndex` to get the index we are looking for in a single pass, rather than using `find` and then `findIndex`.

`import { findIndex } from 'lodash'`, then:

```js
const connect = useCallback(async (type: KeyManager) => {
    dispatch({ type: WalletActions.SET_CONNECTOR_TYPE, payload: type })
    const routeIndex = findIndex(SUPPORTED_WALLETS[type]?.routes, ({ path }) =>
      String(path).endsWith('connect')
    )
    if (routeIndex) {
      dispatch({
        type: WalletActions.SET_INITIAL_ROUTE,
        payload: SUPPORTED_WALLETS[type].routes[routeIndex].path as string
      })
    }
  }, [])

  const create = useCallback(async (type: KeyManager) => {
    dispatch({ type: WalletActions.SET_CONNECTOR_TYPE, payload: type })
    const routeIndex = findIndex(SUPPORTED_WALLETS[type]?.routes, ({ path }) =>
      String(path).endsWith('create')
    )
    if (routeIndex) {
      dispatch({
        type: WalletActions.SET_INITIAL_ROUTE,
        payload: SUPPORTED_WALLETS[type].routes[routeIndex].path as string
      })
    }
  }, [])
```

Reply:
> Nice work.
> 
> Here is a fun little optimisation of that logic for brevity and concision, if you're interested @evandrosaturnino slightly_smiling_face
> 
> We can use `lodash`'s `findIndex` to get the index we are looking for in a single pass, rather than using `find` and then `findIndex`.
> 
> `import { findIndex } from 'lodash'`, then:
> 
> ```js
> const connect = useCallback(async (type: KeyManager) => {
>     dispatch({ type: WalletActions.SET_CONNECTOR_TYPE, payload: type })
>     const routeIndex = findIndex(SUPPORTED_WALLETS[type]?.routes, ({ path }) =>
>       String(path).endsWith('connect')
>     )
>     if (routeIndex) {
>       dispatch({
>         type: WalletActions.SET_INITIAL_ROUTE,
>         payload: SUPPORTED_WALLETS[type].routes[routeIndex].path as string
>       })
>     }
>   }, [])
> 
>   const create = useCallback(async (type: KeyManager) => {
>     dispatch({ type: WalletActions.SET_CONNECTOR_TYPE, payload: type })
>     const routeIndex = findIndex(SUPPORTED_WALLETS[type]?.routes, ({ path }) =>
>       String(path).endsWith('create')
>     )
>     if (routeIndex) {
>       dispatch({
>         type: WalletActions.SET_INITIAL_ROUTE,
>         payload: SUPPORTED_WALLETS[type].routes[routeIndex].path as string
>       })
>     }
>   }, [])
> ```

@0xApotheosis Absolutely, thanks for the optimization. Already implemented it in the WalletProvider, just did a small fix, since 0 is considered false in JS, and the route position for connecting is expected to be as zero, so I changed the conditional validation to `if (routeIndex > -1)`, which is non-false expectancy for this method.

I have run the tests, and everything seems to be good as expected.
Implemented in commit: 421b983319ac3f9f92d473bae0517217ad13d21b